[Shaugn Tan] iP

shogun187 opened the pull request:
## Duke

> All your dreams can come true if you have the courage to pursue them - Walt Disney

Duke frees your mind of having to remember things you need to do. It's,
- text-based
- easy to learn :grinning:

Download Duke by 
1. clicking [here](https://github.com/shogun187/ip).
2. Download the jar file

And it is **FREE!**

**Features:**
- [x] Managing Tasks
- [ ] Managing Deadlines (coming soon)

If you are a **Java programmer**, you can use it to practice Java too. Here's the main method:

```ruby
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

shogun187 committed:
level 1

shogun187 committed:
level 2

shogun187 committed:
level 3

shogun187 committed:
level 4

shogun187 committed:
textUITesting

shogun187 committed:
level 5

shogun187 committed:
level 6

shogun187 committed:
updated level 6

shogun187 committed:
fixed input

shogun187 committed:
removed list

shogun187 committed:
Revert "removed list"

This reverts commit 8d0c803f6053dd40be6f5727e669f5ff820ca25a.

shogun187 committed:
no message

shogun187 committed:
added level 7 stuff

shogun187 committed:
added level 8 stuff

shogun187 committed:
Revert "added level 8 stuff"

This reverts commit 2633a39ba3ac8cb60b390c7245585c46b51ff817.

shogun187 committed:
added level 8 stuff v2

shogun187 committed:
Merge branch 'branch-Level-8'

shogun187 committed:
level 8 v3

shogun187 committed:
Merge branch 'master' into branch-Level-7

shogun187 committed:
no message

shogun187 committed:
Merge remote-tracking branch 'upstream/add-gradle-support'

shogun187 committed:
no message

shogun187 committed:
added gradle support

shogun187 committed:
added more oop

shogun187 committed:
created Duke package

shogun187 committed:
added JUnit tests

shogun187 committed:
no message

shogun187 committed:
no message

shogun187 committed:
Added some javadocs

shogun187 committed:
Added more javadocs

shogun187 committed:
rectified some code to adhere to coding standard

shogun187 committed:
added ability to find tasks with matching words

shogun187 committed:
Merge branch 'branch-Level-9'

shogun187 committed:
Merge branch 'branch-A-JavaDoc'

shogun187 committed:
resolved some conflicts

Bubbl3T commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+package duke;
```
I think it is more advisable to have a blank line after package statements.

Bubbl3T commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+    protected LocalDate time;
+
+    /**
+     * Constructor for a deadline task.
+     * @param description description of task.
+     * @param deadline deadline in YYYY-MM-DD format.
```
Shouldn't there be an empty line between description and parameter section?

Bubbl3T commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+    private TaskList tasks;
+    private Ui ui;
+
+    /**
+     * Constructor for Duke.
+     * @param path filepath of Duke.txt file.
+     */
```
Shouldn't there be an empty line between description and parameter section?

Bubbl3T commented on src/main/java/Duke/Event.java:
```diff
@@ -0,0 +1,33 @@
+    protected String timeString;
+    protected LocalDate time;
+
+    /**
+     * Constructor for an Event.
+     * @param description description of the event.
+     * @param time time of event in YYYY-MM-DD format.
+     */
```
Shouldn't there be an empty line between description and parameter section?

Bubbl3T commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+package duke;
```
I think it is more advisable to have a blank line after package statements.

Bubbl3T commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,121 @@
+        while (ui.hasNextLine()) {
+            String input = ui.getUserCommand();
+            try {
+                //  Bye
```
Great job commenting the different sections for readability! 👍🏻

Bubbl3T commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+ * Deadline class to represent tasks with deadlines.
+ */
+public class Deadline extends Task {
+    protected String deadline;
```
Maybe the variable can be named as `description` instead to avoid confusion with the Deadline object created?

Bubbl3T commented on src/main/java/Duke/Deadline.java:
```diff
@@ -0,0 +1,32 @@
+ */
+public class Deadline extends Task {
+    protected String deadline;
+    protected LocalDate time;
```
I think it may be better to name the variable as `date`, in case you want to use `LocalTime` or `LocalDateTime` in your future commits.

Bubbl3T commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+
+    private TaskList tasklist;
+    private Storage storage;
+    private TaskList tasks;
```
Maybe you will want to comment out or delete this line because I do not see the variable `tasks` used anywhere, and you already have a variable `tasklist` for it.

Bubbl3T commented on src/main/java/Duke/Duke.java:
```diff
@@ -0,0 +1,44 @@
+ */
+public class Duke {
+
+    private TaskList tasklist;
```
You might want to use the camel case for the variable name so it becomes `taskList`. However, leaving it as so is fine and readable too 😄 !

Bubbl3T commented on src/main/java/Duke/EmptyDescriptionException.java:
```diff
@@ -0,0 +1,15 @@
+    public String toString() {
+        return getMessage();
+    }
+}
```
I noticed that some of your files do not end with a new line. I think it is better to do so to make conflict resolution easier.

Bubbl3T commented on src/main/java/Duke/TaskList.java:
```diff
@@ -0,0 +1,99 @@
+package duke;
```
I think it is more advisable to have a blank line after package statements.

Bubbl3T commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,123 @@
+    }
+
+    /**
+     * Returns boolean indicating whether there is further input from the user.
+     * @return true if there is further input, else false.
```
Shouldn't there be an empty line between description and parameter section?

Bubbl3T reviewed (commented):
I think you did great! Just some very minor styling issues here and there. The code is very clean and organized in general too!

shogun187 committed:
fixed more style issues

shogun187 committed:
minor style fix

shogun187 committed:
Update README.md

shogun187 committed:
Update README.md

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
commit before branching level 10

shogun187 committed:
added some lost files

grumphrey commented on src/main/java/Duke/Parser.java:
```diff
@@ -0,0 +1,119 @@
+                    ui.sayBye();
+                    return;
+                }
+                //  List
+                if (input.equals("list")) {
+                    ui.list(tasklist);
+                    continue;
+                }
+                //  Marking
+                if ((input.length() >= 4) && (input.substring(0, 4).equals("mark"))) {
+                    //  error checking
+                    if (input.length() == 4 || input.substring(5).equals("")) {
+                        throw new IllegalIndexException("OOPS!!! The index of a mark cannot be empty.");
+                    }
+                    String remainder = input.substring(5);
+                    int index = Integer.valueOf(remainder) - 1;
+                    tasklist.mark(index, ui);
+                    storage.updateStorage(tasklist);
+                    continue;
+                }
```
I like that you used `continue` instead of if-else blocks. I think it makes the code easier to read!

grumphrey commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,125 @@
+    /**
+     * Tells user that file is not found.
+     */
+    public void fileNotFound() {
+        System.out.println("File not Found!");
+    }
```
```suggestion
    public void printFileNotFound() {
        System.out.println("File not Found!");
    }
```

Perhaps the method name can be changed to a verb instead, e.g. `printFileNotFound()`?

grumphrey commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,125 @@
+     */
+    public void list(TaskList tasklist) {
+        System.out.println("Here are the tasks in your list:");
+        for (int x = 0; x < tasklist.size(); x++) {
+            System.out.println(String.format("%s.%s", x + 1, tasklist.get(x).toString()));
+        }
```
Just an observation, I think its convention to use `i` instead for a loop variable!

grumphrey commented on src/main/java/Duke/Ui.java:
```diff
@@ -0,0 +1,125 @@
+     *
+     * @param tasklist tasklist from Duke application.
+     */
+    public void list(TaskList tasklist) {
```
```suggestion
    public void list(TaskList taskList) {
```
Should it be written in camelCase instead?

grumphrey commented on src/test/java/Duke/DeadlineTest.java:
```diff
@@ -0,0 +1,15 @@
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+
```
```suggestion
```

Extra blank line here!

grumphrey reviewed (commented):
Hey @shogun187 this looks good to me! I don't see much issues besides a few nitpicks. Good job!

shogun187 committed:
added basic GUI

shogun187 committed:
minor style fix

shogun187 committed:
fixed bye input

shogun187 committed:
Merge branch 'branch-Level-10'

shogun187 committed:
Used GFMD

shogun187 committed:
Update README.md

shogun187 committed:
Update README.md

shogun187 committed:
Update README.md

shogun187 committed:
Update README.md

shogun187 committed:
Update README.md

shogun187 committed:
Add assert statements to TaskList class

The TaskList class assumes that the ArrayList has size <=100

The size of the ArrayList may be tampered with

Add assert statements to ensure size of the ArrayList is always <=100

Using assert statements prevent tampering of the ArrayList size

shogun187 committed:
Adhere to coding quality standards

Some parts of the code may be confusing to other readers

Code needs to clearer for easier understanding

Add comments to explain parts of code and removed some dead code

To improve understandability of the code

shogun187 committed:
Merge pull request #2 from shogun187/branch-A-Assertions

A-Assertions

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip into branch-A-CodeQuality

shogun187 committed:
Merge branch 'branch-A-Assertions'

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
Merge pull request #1 from shogun187/branch-A-CodeQuality

A-CodeQuality

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
Merge branch 'branch-A-CodeQuality'

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
Add Command classes and fix Storage to add data/duke.txt file

shogun187 committed:
Remove unused import from Storage

shogun187 committed:
Update README.md

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
Restore help function

shogun187 committed:
Restore help function

shogun187 committed:
Update README.md

shogun187 committed:
Merge branch 'master' of https://github.com/shogun187/ip

shogun187 committed:
Add User Guide and Ui.png

shogun187 committed:
fix Gradle build

shogun187 committed:
Rename Ui.png

shogun187 committed:
Update user guide URL

shogun187 committed:
Add Ui.png

shogun187 committed:
Fix file format for Ui.png

shogun187 committed:
no message

shogun187 committed:
Add Ui.png

shogun187 committed:
Remove unnecessary png files

shogun187 committed:
Update JavaFx version
